client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/task-helper.ts

```diff
@@ -419,11 +432,11 @@ export function getEnhancedTasks(
 	return enhancedTaskList;
 }
 
-function isDomainUpsellCompleted( site: SiteDetails | null ): boolean {
-	return (
-		! site?.plan?.is_free ||
-		site?.options?.launchpad_checklist_tasks_statuses?.domain_upsell_deferred === true
-	);
+function isDomainUpsellCompleted(
+	site: SiteDetails | null,
+	checklistStatuses: LaunchpadStatuses
+): boolean {
+	return ! site?.plan?.is_free || checklistStatuses?.domain_upsell_deferred === true;
```

Comment:
We needed to rollback to checklistStatuses, otherwise only refreshing the page would cause `site.options` to update 🤔 

Our plan is to revert this in a follow-up PR (2295-gh-Automattic/dotcom-forge) so we can release this new flow soon.